tests/test_io/test_index/test_catalog.py

```
@@ -2,6 +2,9 @@
 
 from __future__ import annotations
 
+import threading
+from concurrent.futures import ThreadPoolExecutor, TimeoutError
```

Ruff (0.15.21)

[error] Import `TimeoutError` is shadowing a Python builtin

(A004)